packages/sitecore-jss-react/src/components/PlaceholderCommon.tsx

```diff
@@ -220,6 +228,14 @@ export class PlaceholderCommon<T extends PlaceholderProps> extends React.Compone
       return null;
     }
 
+    // Render SXA Rendering Variant
+    if (renderingDefinition.params?.FieldNames) {
+      return componentFactory(
```

Comment:
Could we use the same `componentModule` concept we are using for component props service? Would take a bit of rework of those types involved, but the benefit is that those are already being generated in the component factory. E.g. here, we would be able to call something like
```suggestion
      const module = await componentModule(renderingDefinition.componentName);
      return module[renderingDefinition.params.FieldNames];
```

Reply:
For me - the main issue here is `async` call, and I can't use it inside the `render` function, we can't wait during that process...

Reply:
And `componentModule` basically used for dynamic import, but it's possible to use `componentFactory` to get anything from the file, but synchronously

Reply:
Yeah, the async/sync complicates re-use. Makes sense to me as-is then.